.travis.yml

```diff
@@ -1,31 +1,46 @@
-sudo: false
+sudo: required
+services:
+  - docker
 language: python
-python:
-  - "2.7"
-addons:
-  postgresql: "9.4"
-before_script:
-  - createdb lumbergh_db
-install:
-  - pip install tox coveralls
 env:
-  - TOX_ENV=flake8
-  - TOX_ENV=docs
-  - TOX_ENV=tests
+  global:
+    # NewRelic
+    - NEWRELIC_STAGE_APPLICATION=careers-stage-us-west
+    - NEWRELIC_PROD_APPLICATION=careers-prod-us-west
+    # Deis
+    - DEIS_STAGE_APPLICATION=careers-stage
+    - DEIS_PROD_APPLICATION=careers-stage
+    - DEIS_USERNAME=travis
+    - DEIS_CONTROLLER=https://deis.us-west.moz.works
+    # Docker
+    - DOCKER_REPOSITORY=mozorg/lumbergh
+    - DOCKER_EMAIL="foo@example.com"
+    - DOCKER_USERNAME="mozjenkins"
+    # Django
+    - DEBUG=False
+    - ALLOWED_HOSTS=*
+    - SECRET_KEY=foo
+    - DATABASE_URL=postgres://postgres@db/postgres
+install:
+  - docker pull ${DOCKER_REPOSITORY}:last_successful_build || true
+  - docker pull ${DOCKER_REPOSITORY}:${TRAVIS_COMMIT} || true
```

Comment:
So here I'm trying to take advantage of docker caching. 

* Before deploying I'm pushing to docker hub with the last_successful_build tag and the git sha tag. 
* During the next travis run (let's say I'm tagging the commit and I'm pushing to production) these lines will download the already built image and docker will skip re-building it and use the cache.
* The `|| true` part of the cmds is there in case there is no `last_successful_build` tag or a tag with the current tested SHA commit. In this case the docker client will return a non zero exit code and travis will stop running the tests when in fact this is not a blocker. Thus I'm adding `|| true` in the end of the line to make sure that the cmd returns successfully regardless docker pull  

For example you can see these sequential runs where I changed `requirements.txt`. The docker layers before `peep install` are all cached.

https://s3.amazonaws.com/archive.travis-ci.org/jobs/103077087/log.txt
https://s3.amazonaws.com/archive.travis-ci.org/jobs/103080065/log.txt